tests/validation/test_highlights.py

```diff
@@ -0,0 +1,366 @@
+        for collection in range(100):
+            response.text = json.dumps(_json)
+            collection_bundle = CollectionBundle(pytest.REQUEST, copy.deepcopy(response), pytest.DURATION,
+                                                 pytest.CONTAINER_STATISTICS)
+            collection_bundle.collection_number = collection
+            growing_unique_objects_overtime.add(collection_bundle)
+            if collection % 3 == 0:
+                _json["result"][0]["key"]["name"] = f"{collection}"
+
+        highlight = highlight_object_growth(growing_unique_objects_overtime.long_collection_statistics)
+        assert (ResultLevel.WARNING,
+                "Object of type HighlightsMP::Growing Object displayed growth of 15.15%") in highlight.messages
+
+    def test_highlight_for_growing_unique_objects_with_low_object_growth(self):
+        response = TestObject()
+        response.is_success = True
+        _json = copy.deepcopy(pytest.JSON)
+        growing_unique_objects_overtime = LongCollectionBundle(5, 25)
+        for collection in range(10):
+            response.text = json.dumps(_json)
+            collection_bundle = CollectionBundle(pytest.REQUEST, copy.deepcopy(response), pytest.DURATION,
+                                                 pytest.CONTAINER_STATISTICS)
+            collection_bundle.collection_number = collection
+            growing_unique_objects_overtime.add(collection_bundle)
+            _json["result"][0]["key"]["name"] = f"{collection}" if collection % 4 == 0 else "same_object"
+
+        highlight = highlight_object_growth(growing_unique_objects_overtime.long_collection_statistics)
+        assert (ResultLevel.WARNING,
+                "Object of type HighlightsMP::Growing Object displayed growth of 6.65%") in highlight.messages
+
+    def test_highlight_for_high_number_of_growing_objects_per_collection(self):
+        response = TestObject()
+        response.is_success = True
+        _json = copy.deepcopy(pytest.JSON)
+        increasing_number_of_objects_per_collection = LongCollectionBundle(5, 25)
+        for collection in range(10):
+            response.text = json.dumps(_json)
+            collection_bundle = CollectionBundle(pytest.REQUEST, copy.deepcopy(response), pytest.DURATION,
+                                                 pytest.CONTAINER_STATISTICS)
+            collection_bundle.collection_number = collection
+            increasing_number_of_objects_per_collection.add(collection_bundle)
+            new_object = copy.deepcopy(pytest.JSON["result"][0])
+            new_object["key"]["name"] = f"{collection}"
+            _json["result"].append(new_object)
+
+        highlight = highlight_object_growth(increasing_number_of_objects_per_collection.long_collection_statistics)
+        assert (ResultLevel.WARNING,
+                "Object of type HighlightsMP::Growing Object displayed growth of 9.65%") in highlight.messages
```

Comment:
This is the other scenario compared to the previous cases, correct? If so we should make sure the messages in each case are different and accurately describe the scenario we are detecting.

Reply:
For posterity: After some discussion, we realized that the case where the number of returned objects grows continuously typically happened in the past due to data structures not being cleared properly between collections. For the SDK, the adapter process does not persist between collections, so this type of error should be much less common. Thus, we are not going to specifically call out this case. If it does happen for some reason, it will still be caught as it is a special case of the general case where the number of unique objects over time is growing.